lib/interfaces.py

```diff
@@ -0,0 +1,83 @@
+import json
+from ops.framework import Object
+
+
+class EtcdProvides(Object):
```

Comment:
Splitting this into a separate file makes the PR noisier than it should be.  I gotta stop doing that.  Will move back for now.